bindings/experimental/distrdf/python/DistRDF/Backends/Base.py

```diff
@@ -170,27 +170,34 @@ def mapper(current_range):
                 # We assume 'end' is exclusive
                 chain.SetCacheEntryRange(start, end)
 
-                # Gather information about friend trees
-                friend_info = current_range.friend_info
-                if friend_info:
-                    # Zip together the treenames of the friend trees and the
-                    # respective file names. Each friend treename can have
-                    # multiple corresponding friend file names.
-                    tree_files_names = zip(
-                        friend_info.friend_names,
-                        friend_info.friend_file_names
+                # Gather information about friend trees. Check that we got an
+                # RFriendInfo struct and that it's not empty
+                if (current_range.friend_info is not None and
+                    not current_range.friend_info.fFriendNames.empty()):
+                    # Zip together the information about friend trees. Each element if of the form:
+                    # (name, alias), (file1.root, file2.root, ...), (subname1, subname2, ...)
+                    zipped_friend_info = zip(
+                        current_range.friend_info.fFriendNames,
+                        current_range.friend_info.fFriendFileNames,
+                        current_range.friend_info.fFriendChainSubNames
                     )
-                    for friend_treename, friend_filenames in tree_files_names:
+                    for (friend_name, friend_alias), friend_filenames, friend_chainsubnames in zipped_friend_info:
                         # Start a TChain with the current friend treename
-                        friend_chain = ROOT.TChain(friend_treename)
+                        friend_chain = ROOT.TChain(str(friend_name))
```

Comment:
this is the price we pay for passing the `RFriendInfo` struct around and not converting it to python strings e.g. in `TreeHeadNode._get_friend_info` . The constructor of TChain (or the `Add` function used a few lines under) accept `const char *), while inside that python loop we are passing around `std::string`s . The only exception are actually `friend_name, friend_alias` which since I unpack them in a pair they probably get converted from C++ strings to Python strings during the unpacking. But I still want to keep this convertion to Python string in all calls inside that loop to avoid having a lot of comments to explain this quirk ( I think the conversion explains the intent much better here).

Reply:
ah it's not a Python problem, it's a C++ problem. nevermind.

Reply:
It goes deeper than I expected actually. So I just tried printing some information from those objects while executing the new roottest added in the sibling PR
```
friend_name: <class 'str'> = 
friend_alias: <class 'str'> = myfriend
friend_filenames: <class cppyy.gbl.std.vector<string> at 0x55f1e974f850> = { "friendtrees_spark.root" }
friend_filenames[0]: <class 'str'> = friendtrees_spark.root
friend_chainsubnames: <class cppyy.gbl.std.vector<string> at 0x55f1e974f850> = { "randomNumbersBis" }
friend_chainsubnames[0]: <class 'str'> = randomNumbersBis
```
And from this it looks like everything is a Python string. But then I get this error
```
TypeError: none of the 2 overloaded methods succeeded. Full details:
  int TChain::Add(TChain* chain) =>
    TypeError: could not convert argument 1
  int TChain::Add(const char* name, Long64_t nentries = TTree::kMaxEntries) =>
    TypeError: could not convert argument 1 (bad argument type for built-in operation)
```
Which is weird becuase if I force the conversion with `str()` I don't get it but it looks like they're python strings anyway...

Reply:
Also the fact that `type(friend_filenames[0])` returns `str` while `type(friend_filenames)` returns `cppyy.gbl.std.vector<string>` is weird. It makes me think that cppyy does something that makes the `std::string` accesed by `friend_filenames[0]` look like a Python str while it's not

Reply:
that's just PyROOT doing its job I'd say?

Reply:
Yes but it doesn't do it all the way down to passing the string as a `const char *` to `TChain::Add`, which instead happens with a Python string created by forced conversion with `str(friend_filenames[0])`. Anyway, probably not an issue for this PR so I'd keep the conversions there

Reply:
I think the confusion here comes from the fact that iterating over an `std::vector<string>` returns `std::string`s, but not accessing elements individually (`friend_filenames[0]` is a Python str). The conversion to Python string you (correctly) added is needed because you get `file_name` from an iteration over the vector.